### DISCUSSION TITLE
No Mixed Integer Support from AMPL optimizer

### DISCUSSION THREAD

**Discussion by Lihan-X** [casadi/casadi#3596] [General]
Hello all, 

I recently want to use the ampl interface to solve a mixed integer problem. Though there is a discrete option, but the default casadi::Nlpsol::integer_support () was not reimplemented. That means I cannot use the ampl solvers to solve mixed integer problems. Is this a fault or there is no support for mixed integer progamming with ampl in casadi yet? (P.S. I want to use some commercial solver in ampl)

best regards, 
Lihan

**Comment by jaeandersson** [comment]
Unless something has changed, the ampl interface to CasADi is quite rudimentary. It also relies on exporting CasADi expressions to AMPL expressions which only works for simple CasADi expressions. So not surprising if it doesn't support integers. Would probably be very easy to add (or fix if it worked in the past).

**Reply by Lihan-X** [reply]
Thank you for your reply. I will try something else then. Have a good day!
